### ISSUE TITLE
Hangs after clicking "Remove OneDrive folder"

### ISSUE THREAD

**Issue by gigjacker** [Beercow/OneDriveExplorer#10]
Hello,

I was relieved and excited to find this project.   OneDrive has been making my life hell for so long and the other day I discovered it wasn't done yet.  I've performed the registry edits suggested in other articles, disabled the service and renamed the folder but my system still thinks those files need a cloud provider and the associate exception is thrown when I try to access them.

The {CIS}.DAT file was loaded successfully and whenI realised the utility isn't going to be able to copy the contents of the files, I right-clicked ont he top level of the tree-view and selected "Remove".   The hourglass appeared and it's been like that ever since, the window fading in that familiar way when a program is not responding.  

I know that the files do have content - they were never cloud-only files.  It's just 1D doing things without asking that has put that folder structure in an inconsistent state.  Surely this utility doesn't require OneDrive to be running in order to remove the folder?

Either way, if you can offer any help as to how I can bypass this interference so that my C# application attempting to move these files can work, it'd be very much appreciated, even if this issue cannot constructively be addressed.

Cheers,
.pd.

**Comment by Beercow** [comment]
I haven’t been able to reproduce the behavior yet. Couple of questions. What version are you running? Are you using the exe or the python script. 
I don’t think this tool is going to do what you want it to. If you are trying to remove OneDrive, this will not work. OneDriveExplorer is a forensics tool to parse OneDrive artifacts from an endpoint and present them in a human readable format.

**Comment by gigjacker** [comment]
Thanks for getting back to me.   I ran the python gui script from the latest zip.  I guess I thought "remove from OneDrive" would unlink the folder for me.

Background: OneDrive has made my life a misery.  I tried unlinking the account and following all the recommended procedures but it would hang indefinitely when I tried switching off on-demand, unlinking folders etc.  Up until on-demand, I was quite happy to let 1D make cloud copies of my files.  But, it seems they weren't content with 50% of my trust and just decided to delete my content and devour 100% of my system resouces maintaining connections that shouldn't have existed in the first place.   

OneDrive is the most evil combo of Trojan Horse and RansomWare in the history of computing.

So what I was really looking for was either a way to bypass the cloud provision check on the command line, a registry hack or even a bit of code.   Failing that, a low-level sub-file i/o method of bypassing the check for cloud provision as "rd /s" seems to be able to do, but instead of deleting the files, I just wanted grab the fragments via whatever on-disk bitmap exists on NTFS and put the files back together in a safe haven far from the intefering clutches of this sickeningly incompetent and unprofessional algorithmic doggeril.  

I've found a C reinvention of rawcopy that I'm investigating, but in the meantime, I've reasoned that since these files in their last accessible state were consistent with the cloud copies, I might as well delete the lot with rd and download anything missing directly from my onedrive account.   That in itself poses a challenge  cuz the download function is cynically unreliable - they don't want you getting hold of your own data and they'll use any dark and sinister means at their disposal to hold your data hostage until you reinstall their amateur excuse for a software product. 

So, now I'm trawling thru onedrive projects on here, looking for functions I can shoplift to get my files via whatever direct API exists - tho, I shudder to think how monumentally awful it'll be when I do find it.

Apologies for the rant.  I really am my wits end.  This has cost me so many days of productivity loss..  If I had the money, I'd sue them.
